src/components/menu/popoverMenu.tsx

```diff
@@ -0,0 +1,138 @@
+import { useMediaQuery } from '@mui/material'
+import Button from '@mui/material/Button'
+import Drawer from '@mui/material/Drawer'
+import IconButton from '@mui/material/IconButton'
+import ListItemIcon from '@mui/material/ListItemIcon'
+import ListItemText from '@mui/material/ListItemText'
+import Menu from '@mui/material/Menu'
+import MenuItem from '@mui/material/MenuItem'
+import { useTheme } from '@mui/material/styles'
+import Typography from '@mui/material/Typography'
+import type { ReactNode } from 'react'
+import React, { useState } from 'react'
+
+export interface PopoverMenuItem {
+  label: string
+  onClick: (event?: React.MouseEvent<HTMLElement>) => void
+  divider?: boolean
+  icon?: ReactNode
+  endDecorator?: ReactNode
+}
+
+export interface PopoverMenuProps {
+  menuItems: PopoverMenuItem[]
+  /** Aria-label to describe the button for assistive technology. */
+  ariaLabel: string
+  /** Provide your own customized icon. */
+  icon?: ReactNode
+  /** Text to display on the button. */
+  buttonText?: string
+}
+
+export default function PopoverMenu(props: PopoverMenuProps) {
+  const [isMenuOpen, setIsMenuOpen] = useState(false)
+
+  const theme = useTheme()
+  const snakeCaseAriaLabel = props.ariaLabel.replace(/ /g, '-')
+  const menuId = `rustic-popover-menu-${snakeCaseAriaLabel}`
+
+  const isMobileView = useMediaQuery(theme.breakpoints.down('md'))
+
+  const PopoverMenuElement = isMobileView ? Drawer : Menu
+
+  let menuBorderStyles: object = {
+    borderRadius: `${theme.shape.borderRadius}px`,
+    border: `1px solid ${theme.palette.divider}`,
+  }
+
+  if (isMobileView) {
+    menuBorderStyles = {
+      ...menuBorderStyles,
+      borderRadius: `${theme.shape.borderRadius}px ${theme.shape.borderRadius}px 0 0`,
+      borderBottom: 0,
+    }
+  }
+
+  const menuItems = props.menuItems.map((menuItem) => {
+    function handleOnClick(event: React.MouseEvent<HTMLElement>) {
+      menuItem.onClick(event)
+      setIsMenuOpen(false)
+    }
+
+    return (
+      <MenuItem
+        key={menuItem.label}
+        onClick={handleOnClick}
+        divider={menuItem.divider}
+      >
+        {menuItem.icon && (
+          <ListItemIcon sx={{ color: 'primary.main' }}>
+            {menuItem.icon}
+          </ListItemIcon>
+        )}
+        <ListItemText>
+          <Typography variant="body2">{menuItem.label}</Typography>
+        </ListItemText>
+        {menuItem.endDecorator && menuItem.endDecorator}
+      </MenuItem>
+    )
+  })
+
+  function handleOpenMenu(event: React.MouseEvent<HTMLElement>) {
+    event.stopPropagation()
+    setIsMenuOpen(true)
+  }
+
+  function renderButton() {
+    const buttonAttributes = {
+      onClick: handleOpenMenu,
+      'aria-label': props.ariaLabel,
+      id: menuId,
+      className: 'rustic-popover-menu-button',
+    }
+
+    if (props.buttonText) {
+      return (
+        <Button
+          {...buttonAttributes}
+          startIcon={props.icon ? props.icon : null}
```

Comment:
I'm wondering if we want to support both start icon and end icon

Reply:
This would be nice. I think we can add this in a follow-up.